src/actions/actions.ts

```diff
@@ -368,6 +368,26 @@ export class ActionPayloadTypes {
         cursor: string
     } = z
 
+    /**
+     * Fetch number of table records
+     * 
+     * @param bcName Business component that initiated data fetch
+     */
+    bcFetchRecordCounter: {
+        bcName: string,
+    } = z
+
+    /**
+     * Fetch number of table records request was succesful
+     * 
+     * @param bcName Business component that initiated data fetch
+     * @param countRecords number of table records
+     */
+    bcFetchRecordCounterSuccess: {
+        bcName: string,
+        countRecords: DataValue,
```

Comment:
It doesn't seem right.  
`DataValue` is the union of possible types for values of data fetching, while you trying to use it as a generic `Record<string, number>` type for your `{ [bcName: string]: number }` structure.
1) You can't just use the unrelated type based on the basis that structurally it matches you goals; you should introduce your own type
2) `DataValue` actually is not a match, that's why you later exlicitly converting it number
3) You probably need here just a regular `number`, it is strange to have a success action with one bc, but with countRecords being a dictionary of results for several bcs